.github/workflows/azure-e2e-test.yml

```diff
@@ -0,0 +1,151 @@
+name: azure-e2e-test
+
+permissions:
+  id-token: write
+  contents: read
+
+env:
+  PODVM_GALLERY_NAME: "podvm_gallery3"
+  PODVM_IMAGE_DEF_NAME: "podvm_image3"
+  PODVM_IMAGE_VERSION: "0.0.3"
+  PODVM_IMAGE_NAME: "peerpod-image-0.0.1"
+  SSH_USERNAME: "peerpod"
+  VM_SIZE: "Standard_D2as_v5"
+
+on:
+  push:
+    branches:
+      - kartikjoshi21/ci-upstream
+
+jobs:
+  build-podvm-image:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: ./.github/actions/setup-azure-build-env
+
+    - name: Clone kata repository
+      run: |
+        pushd ../
```

Comment:
I think each step starts fresh in the job's working dir, the PWD is not kept. so we don't need pushd/popd, maybe `git clone ... -b CCv0 ../kata-containers`?